test/common/ops/changeClass.js

```diff
@@ -27,6 +28,19 @@ describe('shared.ops.changeClass', () => {
     }
   });
 
+  it('req.query.class is an invalid class', (done) => {
+    user.flags.classSelected = false;
+    user.preferences.disableClasses = false;
+
+    try {
+      changeClass(user, {query: {class: 'cellist'}});
+    } catch (err) {
+      expect(err).to.be.an.instanceof(BadRequest);
+      expect(err.message).to.equal(i18n.t('invalidClass'));
+      done();
+    }
```

Comment:
thanks for these suggestions! 

I'm confused about this, though. How does this differ from all other error tests in this same file (or in other test files for that matter). All expect an error to be thrown and only call done() in the catch block. From what I understand the test will return a timeout error anyway, and all error tests seem to behave that way. Should they all be modified in that case?